ptf/saibridgeport.py

```diff
@@ -0,0 +1,381 @@
+
+            print("Checking if FDB entry was removed")
+            status = sai_thrift_set_fdb_entry_attribute(
+                self.client,
+                fdb_entry,
+                packet_action=SAI_PACKET_ACTION_FORWARD)
+            self.assertEqual(status, SAI_STATUS_ITEM_NOT_FOUND)
+
+            print("Sending packet on port %d, %s -> %s - will flood" %
+                  (chck_dev_port, self.dst_mac, self.src_mac))
+            send_packet(self, chck_dev_port, self.chck_tag_pkt)
+            verify_each_packet_on_multiple_port_lists(
+                self, chck_flood_pkt_list, chck_flood_port_list)
+            print("\tOK")
+
+            print("\tVerification complete")
+
+        finally:
+            sai_thrift_flush_fdb_entries(
+                self.client, entry_type=SAI_FDB_FLUSH_ENTRY_TYPE_ALL)
+            sai_thrift_set_bridge_port_attribute(
+                self.client, test_bp, admin_state=True)
+
+    def bpStateDownNoLearnTest(self):
+        '''
+        This checks if MAC addresses are not learned on port when bridge port
+        state is down
+        We set the down state on port0 and send a packet form port0 with given
+        SMAC: 00:11:11:11:11:11, and then we verify it was broadcasted on other
+        ports in VLAN.
+        Next we send a packet from port1 with DMAC 00:11:11:11:11:11
+        and we verify it was also broadcasted on other ports in VLAN
+        '''
+        print("\nbpStateDownNoLearnTest()")
+
+        test_dev_port = self.dev_port0
+        chck_dev_port = self.dev_port1
+        test_bp = self.port0_bp
+        new_bp_port = self.port24
+        new_bp_dev_port = self.dev_port24
+
+        try:
+            # set port0_bp state to DOWN
+            status = sai_thrift_set_bridge_port_attribute(
+                self.client, test_bp, admin_state=False)
+            self.assertEqual(status, SAI_STATUS_SUCCESS)
+            print("Port %d state set to DOWN" % test_dev_port)
+
+            flood_pkt_list = [self.tag_pkt, self.pkt]
+            flood_port_list = [[chck_dev_port],
+                               [self.dev_port4, self.dev_port5,
+                                self.dev_port6]]
+            print("Sending packet on port %d, %s -> %s - will flood" %
+                  (test_dev_port, self.src_mac, self.dst_mac))
+            send_packet(self, test_dev_port, self.pkt)
+            verify_each_packet_on_multiple_port_lists(
+                self, flood_pkt_list, flood_port_list)
+            print("\tOK")
+
+            chck_flood_pkt_list = [self.chck_pkt, self.chck_pkt]
+            chck_flood_port_list = [[test_dev_port],
+                                    [self.dev_port4, self.dev_port5,
+                                     self.dev_port6]]
+            print("Sending packet on port %d with DMAC %s - "
+                  "MAC not learned, will flood" %
+                  (chck_dev_port, self.src_mac))
+            send_packet(self, chck_dev_port, self.chck_tag_pkt)
```

Comment:
It will be better to send the same packet in step1(self.pkt instead of self.check_tag_pkt)

Reply:
The reason for using just this packet is because we want to verify if src MAC address from self.pkt was not learned. If we sent the same packet, verification would be incorrect.